### PR TITLE
Added DCDD from previous gradient

### PR DESCRIPTION
**Pull request by oguevremont**
### Description

Previously, the tracer physics used the current time tracer gradient to compute the DCDD stabilization. Using the previous time gradient instead allows to reduce the number of non-linear iterations while still converging to proper solutions.

### Testing

All tests still pass.

### Documentation

Not applicable.

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [x] All in-code documentation related to this PR is up to date (Doxygen format)
- [x] Lethe documentation is up to date
- [x] New feature has unit test(s) (preferred) or application test(s), and restart files are in the generator folder
- [x] The branch is rebased onto master
- [x] Changelog (CHANGELOG.md) is up to date
- [x] Code is indented with indent-all and .prm files (examples and tests) with prm-indent

Pull request related list:
- [x] Labels are applied
- [x] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [x] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [ ] If the fix is temporary, an issue is opened
- [ ] The PR description is cleaned and ready for merge

### CODE REVIEW

File: source/solvers/tracer_assemblers.cc [hunk 1]
```diff
@@ -43,6 +43,12 @@ TracerAssemblerCore<dim>::assemble_matrix(TracerScratchData<dim> &scratch_data,
       // Store JxW in local variable for faster access;
       const double JxW = JxW_vec[q];
 
+      // We use the previous concentration for the shock capture
+      const Tensor<1, dim> previous_tracer_gradient =
+        scratch_data.previous_tracer_gradients[q];
+
+      const double tracer_gradient_norm = previous_tracer_gradient.norm();
```
Comment (blaisb) [inline source/solvers/tracer_assemblers.cc:50]: This should only be done when the simulation is transient

File: source/solvers/tracer_assemblers.cc [hunk 2]
```diff
@@ -149,24 +156,28 @@ TracerAssemblerCore<dim>::assemble_rhs(TracerScratchData<dim>    &scratch_data,
...
 
       const double vdcdd = (0.5 * h) * (velocity.norm() * velocity.norm()) *
-                           pow(tracer_gradient.norm() * h, order);
+                           pow(tracer_gradient_norm * h, order);
 
       const double   tolerance = 1e-12;
       Tensor<1, dim> s         = velocity / (velocity.norm() + tolerance);
-      Tensor<1, dim> r = tracer_gradient / (tracer_gradient.norm() + tolerance);
+      Tensor<1, dim> r =
+        previous_tracer_gradient / (tracer_gradient_norm + tolerance);
```
Comment (blaisb) [inline source/solvers/tracer_assemblers.cc:180]: Again should only be done for transient simulations

**Review by hepap — COMMENTED** [review]
I don't have anything more to say for this PR :)

A more general comment for DCDD: we have it for VOF, HT and tracer, but the implementation is a bit different in each of them. Maybe in the future we can make it uniform :)

**Review by blaisb — APPROVED** [review]
Great, that should be fine now.

**Review by hepap — APPROVED** [review]

**Pull request merged**